src/fmt/humantime.rs

```diff
@@ -99,13 +95,46 @@ impl fmt::Debug for Timestamp {
 
 impl fmt::Display for Timestamp {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use jiff::fmt::strtime;
+
+        let Ok(ts) = jiff::Timestamp::try_from(self.time) else {
+            return Err(fmt::Error);
+        };
+
         let formatter = match self.precision {
-            TimestampPrecision::Seconds => format_rfc3339_seconds,
-            TimestampPrecision::Millis => format_rfc3339_millis,
-            TimestampPrecision::Micros => format_rfc3339_micros,
-            TimestampPrecision::Nanos => format_rfc3339_nanos,
+            TimestampPrecision::Seconds => |t| strtime::format("%FT%TZ", t),
+            TimestampPrecision::Millis => |t| strtime::format("%FT%T%.3fZ", t),
+            TimestampPrecision::Micros => |t| strtime::format("%FT%T%.6fZ", t),
+            TimestampPrecision::Nanos => |t| strtime::format("%FT%T%.9fZ", t),
+        };
```

Comment:
It's not necessary to use strftime for this. It's likely slower. The `Timestamp` display impl will use rfc 3339 already.

And you can control precision using Rust's standard formatting facilities: https://docs.rs/jiff/latest/jiff/struct.Timestamp.html#example-35

Reply:
Thanks! That was an awesome tip. Fixed!